test/extended/etcd/leader_changes.go

```diff
@@ -0,0 +1,26 @@
+package etcd
+
+import (
+	"context"
+	"time"
+
+	g "github.com/onsi/ginkgo"
+	o "github.com/onsi/gomega"
+	"github.com/prometheus/common/model"
+
+	"github.com/openshift/origin/test/extended/prometheus/client"
+	exutil "github.com/openshift/origin/test/extended/util"
+)
+
+var _ = g.Describe("etcd", func() {
+	defer g.GinkgoRecover()
+	oc := exutil.NewCLI("etcd-leader-change", exutil.KubeConfigPath())
+	g.It("leader changes are not excessive", func() {
+		prometheus, err := client.NewE2EPrometheusRouterClient(oc)
+		o.Expect(err).ToNot(o.HaveOccurred())
+		g.By("Examining the rate of increase in the number of etcd leadership changes for last five minutes")
+		result, _, err := prometheus.Query(context.Background(), "increase((max by (job) (etcd_server_leader_changes_seen_total) or 0*absent(etcd_server_leader_changes_seen_total))[15m:1m])", time.Now())
+		o.Expect(err).ToNot(o.HaveOccurred())
+		o.Expect(result.(model.Vector)[0].Value).To(o.BeNumerically("==", 0))
```

Comment:
@sanchezl nicely readable :+1:

Reply:
Does this mean that the rate is zero?

Reply:
And why only the first element?

Reply:
Only one value is returned.

Reply:
Still open: Does this mean that the rate is zero?

Everywhere else where this code is used, we compare with `3`, not zero.

Reply:
The query was suggested in https://github.com/openshift/origin/pull/24291#discussion_r359007429.
I think this is saying the rate of increase over the last five minutes is `0`. 
This should take care of the issue brought up in https://github.com/openshift/origin/pull/24291#discussion_r358430475 where clusters on some cloud providers might have some different levels of etcd thrashing before settling down.  We use to compare the `etcd_server_leader_changes_seen_total` to `3`, but that value was only know to be appropriate on AWS (and even there, some preferred a stricter value of `1`).